### PR TITLE
refactor: hide publication date

### PR DESCRIPTION
**Pull request by sshanzel**
DD-340 #done

The separator now has 2 conditions for whether to be displayed or not.

The article publication date should also now be depending on our flagsmith value.

### CODE REVIEW

**Comment by sshanzel** [comment]
Also, I noticed the max height for post card should actually be `366px`.

image

File: packages/shared/src/components/cards/PostCard.spec.tsx [hunk 1]
```diff
@@ -107,10 +117,23 @@ it('should call on bookmark click on bookmark button click', async () => {
   );
 });
 
+it('should not display publication date based on features flags', async () => {
+  act(async () => {
```
Comment (idoshamun) [inline packages/shared/src/components/cards/PostCard.spec.tsx:121]: Why do we need the `act` function here?

File: packages/shared/src/components/cards/PostCard.spec.tsx [hunk 2]
```diff
@@ -107,10 +117,23 @@ it('should call on bookmark click on bookmark button click', async () => {
...
+    const el = await screen.findByText('Jun 13, 2018');
+    expect(el).not.toBeInTheDocument();
+  });
+});
+
 it('should format publication date', async () => {
-  renderComponent();
-  const el = await screen.findByText('Jun 13, 2018');
-  expect(el).toBeInTheDocument();
+  act(async () => {
```
Comment (idoshamun) [inline packages/shared/src/components/cards/PostCard.spec.tsx:132]: Why do we need the `act` function here?

File: packages/shared/src/components/cards/PostMetadata.tsx [hunk 3]
```diff
@@ -12,7 +25,9 @@ export default function PostMetadata({
   className?: string;
   children?: ReactNode;
 }): ReactElement {
+  const { flags } = useContext(FeaturesContext);
```
Comment (idoshamun) [inline packages/shared/src/components/cards/PostMetadata.tsx:28]: Exposing this component to context and especially to the `FeaturesContext` will create tight coupling between them. I think it's better to accept this flag as property from the outside.

**Review by idoshamun — CHANGES_REQUESTED** [review]
A few fixes and we can ship :P

File: packages/shared/src/components/cards/PostCard.spec.tsx (hunk 1 above)
Comment (sshanzel) [reply packages/shared/src/components/cards/PostCard.spec.tsx]: I think some re-renders are triggered, and the tests were failing. After I applied the `act` to signify that rendering might take time, it worked.

Comment (sshanzel) [reply packages/shared/src/components/cards/PostCard.spec.tsx]: Ah, my procedure was just wrong, which makes the act necessary.

**Review by idoshamun — APPROVED** [review]
Awesome!

**Review by rebelchris — APPROVED** [review]
Awesome work Lee!
And believe this is the first introduction of flagsmith on the monorepo right?

Looking forward to doing more of these little UI tests.

Subnote:
Had a giggle on their naming convention:
`IBulletTrainFeature`

**Comment by idoshamun** [comment]
@rebelchris flagsmith was already there in several places but not that common

**Comment by sshanzel** [comment]
Thank you for the reviews both!

**Comment by sshanzel** [comment]
The feat has now been disabled. I will be merging this now. Thank you!

image

**Pull request merged**